### ISSUE TITLE
IDE crashing

### ISSUE THREAD

**Issue by pvc28** [MiniZinc/MiniZincIDE#75]
Running MiniZinc 2.2.3 (x64) on Windows10 Pro (Version 1803 OS).  Can open the IDE, enter program manually and run.  BUT cannot Save the file, or open a file using the menu in the IDE - it will just immediately crash - every time.

If I want to open or save a file, it will open the window to let me choose a file to open or give it a filename to save but then crash approx 5 seconds later.

I can open a file by double-clicking on the file in file explorer - but then cannot save it again - the IDE crashes.

If I enter text manually and run the program, it will work a few times and then randomly crash.

Seems similar to the problem #48.  The problem seems to be that it is crashing when it has to interface with the file system in windows - opening or saving .mzn or data files.

Any solution?

**Comment by pvc28** [comment]
Anyone...?

**Comment by guidotack** [comment]
I can't reproduce the problem here. Is there anything special about the file system on your machine, e.g. do you have unicode characters in the path names, what language is your OS set to, is it a network file system, anything like that "out of the ordinary" that could explain why it doesn't crash here?

**Comment by pvc28** [comment]
Hi,
There's nothing unusual about the machine at all.  It's a vanilla English Language ordinary windows 10 machine.  It runs all my other software with no problem - Word, Excel, Adobe Premiere, Photoshop, email, Chrome, video software etc without any problem.  I'm running Norton Antivirus and it's clean.  It's got heaps of memory and the problems persists even if I restart the machine and run only Minizinc - so it's not a problem with memory.  It it is an unmodified Dell computer with a 2TB hard drive with 1.36 TB free, 12 GB RAM, an i7-3770 CPU running at 3.4 GHz, x64 processor; running Windows 10 Pro Version 1803.  I'm not the only one who has had this problem.  Reading this forum and the coursera forums, there are a couple of people who describe similar problems.  One person solved it by using the 32 but version instead but of course I can't do that now.

**Comment by pvc28** [comment]
I have some avi screen capture files that show it crashing if that would help.  But this site won't let me attach them.  So I can email them if you give me an address.

**Comment by pvc28** [comment]
OK - here is a zip file with three avi files showing how it crashes.
crash when opening file   - try to open a file and it hangs for approx 5 seconds (I can move the cursor but clicking on the open button gets no response ) then the window just closes and it crashes

crash when save1 - it opens the save box but then hangs and closes after approx 5 seconds

crash when save 2 - if I try to type in a save filename, it hangs for approx 5 seconds - will take no input, then it shows the characters I typed for the filename briefly before crashing

[crash videos.zip](https://github.com/MiniZinc/MiniZincIDE/files/2785316/crash.videos.zip)

**Comment by Dekker1** [comment]
These crashes all seem to be related to the file selector, for which we depend on the Qt library. We'll have to investigate if this is a problem with the Qt version that was used for the latest release.

**Comment by Dekker1** [comment]
@pvc28 Since I cannot recreate the bug on my windows environment. Would you mind testing the current development build? I've uploaded it to google drive: https://drive.google.com/file/d/1wIgm5FKypQCdnDBdnXw7bejan2tvxs_p/view?usp=sharing

This version has been compiled using Qt5.9.7 and MSVC 2017 (instead of Qt5.9.6 and MSVC 2015). Hopefully this will solve the problem. The alternative would be to try and use the new LTS version of  Qt: 5.12

**Comment by pvc28** [comment]
I downloaded, installed and ran the development build.  Still have the same problem.  I have run every windows update available - made no difference.  Did a system file check - found nothing wrong.  Have tried rebooting the computer and running minizinc without any other programs open - made no difference. 
 On one occasion it did actually work for a few minutes and I was able to open a few files - but then it crashed - and I couldn't reproduce it so I don't know why I got a bit further that time.  

While clearly not everyone has this problem, some others have described what appears to be a similar issue over the past year - but I can't figure out why are we having this issue and nobody else?  

Not sure what you mean by this: The alternative would be to try and use the new LTS version of Qt: 5.12
- are you referring to you using Qt 5.12?  It is supposed to fix more than 2000 bugs (!) since 5.9.7.

![crash1](https://user-images.githubusercontent.com/38054754/51645560-08522700-1fc9-11e9-8b3e-ca584ec67aa2.jpg)

![crash2](https://user-images.githubusercontent.com/38054754/51645561-0be5ae00-1fc9-11e9-9a86-a3533bbd6957.jpg)

![crash3](https://user-images.githubusercontent.com/38054754/51645565-0f793500-1fc9-11e9-9b8f-f34b15f279ea.jpg)
